scala-thoughts.md

```diff
@@ -0,0 +1,37 @@
+### What's easy to do in Scala? What's not?
+* I was impressed with the variety of methods on lists! I used map and filter, because those
+were methods that I learned in my CS52 class last semester. 
+One of Scala's greatest strengths is how concise it can be,
+after you learn how to properly chain methods. 
+
+### What is/are your favorite language design choice(s) that the designers of Scala made? Why?
+* To be honest, I found this assignment to be quite difficult... but here are some of the cool things that I observed.
+* I don't know whether I like this, but I think it's interesting that Scala is designed so that a "phrase of code" doesn't need
+to end in a semicolon. I think it's nice that having it doesn't hurt. Maybe this is because Scala was designed to cater to the 
+coding habits of a lot of different users who are used to different programming languages.
+* Even though I had to get used to Scala's way of constructing method headers, I think I like them! For example,
+def isPalindrome(str: String): Boolean = { ... } makes sense to me. I especially like "Boolean = { ... }" because it makes it 
+clear that whatever is returned by the code between the curly brackets is of type Boolean. 
+
+### What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+* One of my least favorite design choices is how to print a string. Perhaps it's because I'm used to how it's done in Java, 
+but typing the following: s"A string with length ${str.length}" was so tedious! I much prefer writing "A string with length " 
++ str.length . I'm not sure why this choice was made...
+* I also do not like how you cannot reassign the value of a var. It was annoying to want to type "i++" somewhere, and then realize

```

Comment:
`val` is the immutable variable, while `var` can be changed. I do believe it's difficult for imperative programmers to get used to immutability (which is the main goal for functional programming). I guess in other languages, you don't have the options to choose (i.e. all variables are mutable), but in creating many options, Scala must differentiate between immutable and mutable variables.